cpp/daal/src/externals/config_template.h

```diff
@@ -0,0 +1,6 @@
+#ifndef __CONFIG_H__
+#define __CONFIG_H__
+
+#include "src/externals/config_{BACKEND}.h"
```

Comment:
Should we just change this approach to use defines instead of relaying on sed for replacement?

Reply:
The idea is to avoid too complex rules for defines and provide a single entry point. But it might be possible to use -DDAAL_REF for that. In first case, only name of backed (existing one) should be provided for make and everything else will taken from config. Second option is to set backend_name --> define mapping in makefile and then use define macro inside header files for as a switch. In first case, mistake in a particular backend will not affect other ones.